src/core/MIGRATION_EXAMPLES.md

```diff
@@ -0,0 +1,255 @@
+which gives full access to all of Hapi's API. In the New Platform, plugins
+have access to the [HttpServiceSetup]() interface, which is exposed via the
+[CoreSetup] object injected into the `setup` method of server-side plugins.
+
+This interface has a different API with slightly different behaviors.
+- All input (body, query parameters, and URL parameters) must be validated using
+  the `@kbn/config-schema` package. If no validation schema is provided, these
+  values will be empty objects.
+- All exceptions thrown by handlers result in 500 errors. If you need a specific
+  HTTP error code, catch any exceptions in your handler and construct the
+  appropriate response using the provided response factory.
+
+### Route Registration
+
+```ts
+// Legacy route registration
+import Joi from 'joi';
+
+new kibana.Plugin({
+  init(server) {
+    server.route({
+      path: '/api/my-plugin/my-route',
+      method: 'POST',
+      options: {
+        validate: {
+          payload: Joi.object({
+            field1: Joi.string().required(),
+          }),
+        }
+      },
+      async handler(req, h) {
+        return { message: `Received field1: ${req.payload.field1}` };
+      }
+    });
+  }
+});
+
+
+// New Platform equivalent
+import { schema } from '@kbn/config-schema';
+
+class Plugin {
+  public setup(core) {
+    const router = core.http.createRouter();
+    router.post(
+      {
+        path: '/api/my-plugin/my-route',
+        validate: {
+          body: schema.object({
+            field1: schema.string(),
+          }),
+        }
+      },
+      async (context, req, res) => {
+        return res.ok({
+          body: {
+            message: `Received field1: ${req.body.field1}`
+          }
+        });
+      }
+    )
+  }
+}
+```
+
+### Accessing Services
+
+Services in the Legacy Platform were typically available via methods on either
+`server.plugins.*`, `server.*`, or `req.*`. In the New Platform, all services
+are available via the `context` argument to the route handler. The type of this
+argument is the [RequestHandlerContext](). The APIs available here will include
+all Core services and any services registered by plugins this plugin depends on.
+
+```ts
+new kibana.Plugin({
+  init(server) {
+    const { callWithRequest } = server.plugins.elasticsearch.getCluster('data');
+
+    server.route({
+      path: '/api/my-plugin/my-route',
+      method: 'POST',
+      async handler(req, h) {
+        const results = await callWithRequest(req, 'search', query);
+        return { results };
+      }
+    });
+  }
+});
+
+class Plugin {
+  public setup(core) {
+    const router = core.http.createRouter();
+    router.post(
+      {
+        path: '/api/my-plugin/my-route',
+      },
+      async (context, req, res) => {
+        const results = await context.elasticsearch.dataClient.callAsCurrentUser('search', query);
+        return res.ok({
+          body: { results }
+        });
+      }
+    )
+  }
+}
+```
+
+## Chrome
```

Comment:
Shouldn't we separate examples by server-path / client-path?

Reply:
Not sure what you mean. I don't believe we have any chrome APIs on the server? Or are you talking about basePath APIs?

Reply:
It's weird to jump to `Chrome` without referencing to the client side.
I'd even split the manual into 2 different documents to be more focused.